src/loader.js

```diff
@@ -189,45 +189,49 @@ export function pitch(request) {
   childCompiler.runAsChild((error, entries, compilation) => {
     const assets = Object.create(null);
     const assetsInfo = new Map();
+    const emit = typeof options.emit !== 'undefined' ? options.emit : true;
 
-    for (const asset of compilation.getAssets()) {
-      assets[asset.name] = asset.source;
-      assetsInfo.set(asset.name, asset.info);
-    }
-
-    const addDependencies = (dependencies) => {
-      if (!Array.isArray(dependencies) && dependencies != null) {
-        throw new Error(
-          `Exported value was not extracted as an array: ${JSON.stringify(
-            dependencies
-          )}`
-        );
+    let addDependencies;
+    if (emit) {
+      for (const asset of compilation.getAssets()) {
+        assets[asset.name] = asset.source;
+        assetsInfo.set(asset.name, asset.info);
       }
 
-      const identifierCountMap = new Map();
+      addDependencies = (dependencies) => {
+        if (!Array.isArray(dependencies) && dependencies != null) {
```

Comment:
This doesn't get covered by test and can be removed. The only caller of this function is a few lines below and is guaranteed to be an array.

Should I remove it?